src/rust-crypto/rust-crypto.ts

```diff
@@ -1754,23 +1755,24 @@ class EventDecryptor {
         private readonly perSessionBackupDownloader: PerSessionKeyBackupDownloader,
     ) {}
 
-    public async attemptEventDecryption(event: MatrixEvent, cryptoMode: CryptoMode): Promise<IEventDecryptionResult> {
+    public async attemptEventDecryption(
+        event: MatrixEvent,
+        isolationMode: DeviceIsolationMode,
+    ): Promise<IEventDecryptionResult> {
         // add the event to the pending list *before* attempting to decrypt.
         // then, if the key turns up while decryption is in progress (and
         // decryption fails), we will schedule a retry.
         // (fixes https://github.com/vector-im/element-web/issues/5001)
         this.addEventToPendingList(event);
 
         let trustRequirement;
-        switch (cryptoMode) {
-            case CryptoMode.Legacy:
-                trustRequirement = RustSdkCryptoJs.TrustRequirement.Untrusted;
-                break;
-            case CryptoMode.Transition:
+
+        switch (isolationMode.kind) {
+            case "OnlySignedIsolation":
                 trustRequirement = RustSdkCryptoJs.TrustRequirement.CrossSignedOrLegacy;
```

Comment:
`CrossSignedOrLegacy` is the mode that we want. `CrossSigned` only would be for a more extreme mode that we don't have yet.